### PR TITLE
working tested gcp oses

### PR DESCRIPTION
**Pull request by bernadinm**
* added license
* added dcos-tested-azure-oses working module
* added WIP documentation (not functional yet, but available)

### CODE REVIEW

File: gcp/modules/dcos-tested-gcp-oses/platform/cloud/gcp/centos_7.2/setup.sh [hunk 1]
```diff
@@ -0,0 +1,35 @@
+sudo setenforce 0 && \
```
Comment (mnaboka) [inline gcp/modules/dcos-tested-gcp-oses/platform/cloud/gcp/centos_7.2/setup.sh]: should not a shell script start with shebang `#!/bin/sh` ?

File: gcp/modules/dcos-tested-gcp-oses/platform/cloud/gcp/centos_7.3/setup.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
+sudo setenforce 0 && \
```
Comment (mnaboka) [inline gcp/modules/dcos-tested-gcp-oses/platform/cloud/gcp/centos_7.3/setup.sh]: same here #!/bin/sh

File: gcp/modules/dcos-tested-gcp-oses/platform/cloud/gcp/rhel_7.3/setup.sh [hunk 3]
```diff
@@ -0,0 +1,32 @@
+sudo setenforce 0 && \
```
Comment (mnaboka) [inline gcp/modules/dcos-tested-gcp-oses/platform/cloud/gcp/rhel_7.3/setup.sh]: same here #!/bin/sh

File: gcp/modules/dcos-tested-gcp-oses/platform/cloud/gcp/centos_7.3/setup.sh [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+sudo systemctl enable docker
+sudo systemctl stop dnsmasq.service
+sudo systemctl disable dnsmasq.service
+sudo yum install -y wget
+sudo yum install -y git
+sudo yum install -y unzip
+sudo yum install -y curl
+sudo yum install -y xz
+sudo yum install -y ipset
+sudo yum install -y ntp
```
Comment (mnaboka) [inline gcp/modules/dcos-tested-gcp-oses/platform/cloud/gcp/centos_7.3/setup.sh]: while i am not a bash expert, would not it be easier to list all dependancies in one place, and then
install all of them with a command
`sudo yum install -t <command1 command2 command3>` ?

File: gcp/modules/dcos-tested-gcp-oses/main_variables.tf [hunk 5]
```diff
@@ -0,0 +1,6 @@
+variable "provider" {
+ default = "gcp"
+}
+
+variable "os" {}
+variable "region" {}
```
Comment (mnaboka) [inline gcp/modules/dcos-tested-gcp-oses/main_variables.tf]: is there a reason `os` and `region` are empty by default?

File: gcp/modules/dcos-tested-gcp-oses/main_template_file.tf [hunk 6]
```diff
@@ -0,0 +1,4 @@
+# Cloud Image Instruction
+data "template_file" "os-setup" {
+template = "${file("${path.module}/platform/cloud/${var.provider}/${var.os}/setup.sh")}"
```
Comment (mnaboka) [inline gcp/modules/dcos-tested-gcp-oses/main_template_file.tf]: nit: ident

File: gcp/modules/dcos-tested-gcp-oses/input_test_profile [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+# azure_publisher = RedHat
+# azure_sku = 7.3
+# azure_version = 7.3.2017053118
+# ------------------------------
+provider = "azure"
+os = "rhel_7.3"
+#os = "centos_7.3"
+#os = "coreos_835.13.0"
+#os = "coreos_1235.9.0"
+region = "westus"
```
Comment (mnaboka) [inline gcp/modules/dcos-tested-gcp-oses/input_test_profile]: not sure why some of them commented out, can we make it a little bit more consistent? do we even need this checked in?

File: gcp/modules/dcos-tested-gcp-oses/platform/cloud/gcp/centos_7.3/setup.sh (hunk 4 above)
Comment (bernadinm) [reply gcp/modules/dcos-tested-gcp-oses/platform/cloud/gcp/centos_7.3/setup.sh]: This section was reused by this portion of DC/OS here: https://github.com/dcos/dcos/blob/master/dcos_installer/action_lib.py#L431-L436

You are correct, we can make this in one line. I assumed that readability was in a higher priority over completeness. 

Let me know which one you perfer in this case.

File: gcp/modules/dcos-tested-gcp-oses/input_test_profile (hunk 7 above)
Comment (bernadinm) [reply gcp/modules/dcos-tested-gcp-oses/input_test_profile]: This file is useful for manual testing but can be removed. Proper CI jobs would need to be made for this repo to ensure integrity of the modules.

File: gcp/modules/dcos-tested-gcp-oses/main_variables.tf (hunk 5 above)
Comment (bernadinm) [reply gcp/modules/dcos-tested-gcp-oses/main_variables.tf]: the actual values of `os` and `region` are populated and required by the top level root module github.com/dcos/terraform-dcos/aws github.com/dcos/terraform-dcos/azure, etc. 

By listing the values of os and region would be inapplicable.

**Pull request merged**